src/components/PanelV2/PanelV2.js

```diff
@@ -116,8 +117,9 @@ function PanelV2({
       <Transition className={namespace}>
         {isOpen && (
           <Portal
-            focusTrap={focusTrap}
+            focusTrap={withOverlay || focusTrap}
             focusTrapOptions={focusTrapOptions}
+            hasOverlay={withOverlay}
```

Comment:
for the sake of consistency with the `focusTrap` prop might wanna go without the `has` prefix

Reply:
I was following https://github.com/carbon-design-system/carbon/blob/master/packages/react/src/components/Loading/Loading.js#L105 as a reference point, but we can align with the individual component props if that makes more sense